### PR TITLE
LPS-122393 Check permissions for bulk actions

### PR DESCRIPTION
**Pull request by robertoDiaz**
Related Issue: https://issues.liferay.com/browse/LPS-122393

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @robertoDiaz, can you please check if these failures are related to your changes? Thanks

**Comment by robertoDiaz** [comment]
I think that this failures are not related:

Ones are in journal, and I do not touch this in this PR

The other one seem to be in the export import dialog in forms.

What I do in this PR is disable the remove button when the user is not able to remove form entry records, so I cant' see how they could be connected.

Regards

**Comment by robertoDiaz** [comment]
Hi.

Whats the status of this?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/form_instance_records_search_container.jsp [hunk 1]
```diff
@@ -62,6 +62,11 @@ PortletURL portletURL = ddmFormViewFormInstanceRecordsDisplayContext.getPortletU
 
 				Map<String, DDMFormField> ddmFormFieldsMap = ddmForm.getDDMFormFieldsMap(true);
 
+				row.setData(
+					HashMapBuilder.<String, Object>put(
+						"actions", StringUtil.merge(ddmFormViewFormInstanceRecordsDisplayContext.getAvailableActions(permissionChecker))
+					).build());
+
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/form_instance_records_search_container.jsp:69]: @robertoDiaz, during your investigation, have you considered tackle this issue [here](https://github.com/liferay-forms/liferay-portal/blob/5744de7cb61ff47fd607a04cc09d7b0a7f5e9b4b/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/form_instance_record_action.jsp#L62) instead?

**Comment by robertoDiaz** [comment]
Hi @natocesarrego 

I've followed the pattern used in the whole portal to handle the bulk options in the search container, so I think is the approach we should use in this case. See this examples (there are more):

```
com.liferay.announcements.web.internal.display.context.AnnouncementsAdminViewManagementToolbarDisplayContext#getAvailableActions
com.liferay.asset.categories.admin.web.internal.display.context.AssetCategoriesManagementToolbarDisplayContext#getAvailableActions 
com.liferay.asset.categories.admin.web.internal.display.context.AssetVocabulariesManagementToolbarDisplayContext#getAvailableActions
```
Anyway I do not fully understand your answer. Maybe we could use the isShowDeleteIcon method, but nothing more.

BTW, a customer is waiting for this fix, so, can you prioritize this? If you find that there is a better solution, feel free to push it to master and I'll use it.

Thanks!

**Comment by natocesarrego** [comment]
Hi @robertoDiaz,

Thank you very much for your clarification! 🙂 I believe your solution is good to go, however I've one last question to you:

When the bulk delete operation isn't available then no other bulk operation will be available, in light of this how the UI will handle it? Does make sense keep the checkbox for bulk operations?

Thanks.

**Comment by robertoDiaz** [comment]
What this logic does is check if the user is able to delete any of the selected entries.

If able, it includes it in an array (and discards the ones the user can´t handle)

If the user can´t delete any of the selected entries, disables the button.

As examplo of a multiple bulk operation mechanism where each bulk operation is handled by separated you could see `com.liferay.document.library.web.internal.display.context.DLAdminManagementToolbarDisplayContext#getAvailableActions(com.liferay.portal.kernel.repository.model.FileEntry)
`
Here we have checks for each action:

```
ActionKeys.DELETE permission check for deleteEntries
ActionKeys.UPDATE permission check for move, checkout, checkin
A conditional for editCategories or editTags
ActionKeys.VIEW permission check for Download
```

So each action is enabled when one of them is added to the availableActions List.

As said is a general pattern.

About your last question: This behavior is a UX decision, and from my perspective, we should be consistent, so I think that yes, it makes sense to keep the checkbox.

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
ci:test:sf

**Comment by robertoDiaz** [comment]
ci:test:stable

**Comment by liferay-forms** [comment]
Hey @robertoDiaz, can you please fix the source formatter issues and resend? Thanks

**Comment by natocesarrego** [comment]
Hey @robertoDiaz, after read your last comment, I understood that if the user doesn't have permission to delete an entry, then the delete button will be disabled. If my understanding is correct, then I'm ready to move on with this pull. 🙂 However, I believe you did some update on it which generated a SF issue causing the pull to be automatically closed. Therefore, when you're ready again, please send a new pull.

Thanks.

**Comment by robertoDiaz** [comment]
Yes, that's correct.

I did a change because the pull conflicted with the master branch (I send it a week ago)

I'm going to update it, but please, in the future try to move the PR a little bit faster to avoid this conflicts (I know this kind of  mechanisms very well: I apply them to most of the Collaboration portlets when I was in the engineering team) 

Thanks 🍺

**Pull request closed without merging**